crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -195,10 +195,13 @@ def prepare_nodes_for_crowbar_upgrade
     # To all nodes, add a new role which prepares them for the upgrade
     nodes_to_upgrade = []
     all_nodes = NodeObject.all
-    not_ready = all_nodes.select { |n| !n.admin? && n.state != "ready" }.map { |n| n.name }
+    not_ready_for_upgrade = all_nodes.select { |n| !n.admin? &&
+                                    n.state != "ready" &&
+                                    n.state != "crowbar_upgrade"
```

Comment:
I would have written: `!%w(ready crowbar_upgrade).include?(n.state)`

But it's fine to keep it the current way.